FirebaseAI/Sources/GenerateContentResponse.swift

```diff
@@ -513,3 +639,105 @@ extension PromptFeedback: Decodable {
     }
   }
 }
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+extension GroundingMetadata: Decodable {
+  enum CodingKeys: String, CodingKey {
+    case webSearchQueries
+    case groundingChunks
+    case groundingSupports
+    case searchEntryPoint
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    webSearchQueries = try container.decodeIfPresent([String].self, forKey: .webSearchQueries) ?? []
+    groundingChunks = try container.decodeIfPresent(
+      [GroundingChunk].self,
+      forKey: .groundingChunks
+    ) ?? []
+    groundingSupports = try container.decodeIfPresent(
+      [GroundingSupport].self,
+      forKey: .groundingSupports
+    ) ?? []
+    searchEntryPoint = try container.decodeIfPresent(
+      SearchEntryPoint.self,
+      forKey: .searchEntryPoint
+    )
+  }
+}
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+extension GroundingMetadata.SearchEntryPoint: Decodable {
+  enum CodingKeys: String, CodingKey {
+    case renderedContent
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    renderedContent = try container.decodeIfPresent(String.self, forKey: .renderedContent)
+  }
+}
```

Comment:
Since this implementation doesn't have any custom decoding logic (unlike the `GroundingMetadata` one above), let's just do `extension GroundingMetadata.SearchEntryPoint: Decodable {}`. That way the `Decodable` implementation will get auto-synthesized.

Reply:
Done for all where applicable